src/sentry/dynamic_sampling/tasks.py

```diff
@@ -0,0 +1,100 @@
+from typing import Sequence, Tuple
+
+from sentry import features, quotas
+from sentry.dynamic_sampling.models.adjustment_models import AdjustedModel
+from sentry.dynamic_sampling.models.adjustment_models import DSProject as DSProject
+from sentry.dynamic_sampling.prioritise_projects import fetch_projects_with_total_volumes
+from sentry.dynamic_sampling.rules.helpers.prioritise_project import _generate_cache_key
+from sentry.dynamic_sampling.rules.utils import OrganizationId, ProjectId, get_redis_client_for_ds
+from sentry.models import Organization, Project
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.relay import schedule_invalidate_project_config
+from sentry.utils import metrics
+
+CHUNK_SIZE = 1000
+MAX_SECONDS = 60
+CACHE_KEY_TTL = 24 * 60 * 60 * 1000
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.prioritise_projects",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=2 * 60 * 60,  # 2hours
+    time_limit=2 * 60 * 60 + 5,
+)  # type: ignore
+def prioritise_projects() -> None:
+    metrics.incr("sentry.tasks.dynamic_sampling.prioritise_projects.start", sample_rate=1.0)
+    with metrics.timer("sentry.tasks.dynamic_sampling.prioritise_projects", sample_rate=1.0):
+        for org_id, projects_with_tx_count in fetch_projects_with_total_volumes().items():
+            process_projects_sample_rates.delay(org_id, projects_with_tx_count)
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.process_projects_sample_rates",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=25 * 60,  # 25 mins
+    time_limit=2 * 60 + 5,
+)  # type: ignore
+def process_projects_sample_rates(
+    org_id: OrganizationId, projects_with_tx_count: Sequence[Tuple[ProjectId, int]]
+) -> None:
+    """
+    Takes a single org id and a list of project ids
+    """
+    organization = Organization.objects.get_from_cache(id=org_id)
+    # Check if feature flag is enabled:
+    if features.has("organizations:ds-prioritise-by-project-bias", organization):
+        with metrics.timer("sentry.tasks.dynamic_sampling.process_projects_sample_rates.core"):
+            adjust_sample_rates(org_id, projects_with_tx_count)
+
+
+def adjust_sample_rates(
+    org_id: int, projects_with_tx_count: Sequence[Tuple[ProjectId, int]]
+) -> None:
+    """
+    This function apply model and adjust sample rate per project in org
+    and store it in DS redis cluster, then we invalidate project config
+    so relay can reread it, and we'll inject it from redis cache.
+    """
+    projects = []
+    project_ids_with_counts = {}
+    for project_id, count_per_root in projects_with_tx_count:
+        project_ids_with_counts[project_id] = count_per_root
+
+    for project in Project.objects.get_many_from_cache(project_ids_with_counts.keys()):
+        sample_rate = quotas.get_blended_sample_rate(project)
```

Comment:
Let's please closely watch the performance of this loop. `get_blended_sample_rate` in getsentry loads the Subscription model without cache access; this may be slow for a large number of projects. We may want to introduce another batch function for this (that defaults to looping) and then implement this in `getsentry/quotas`.

Reply:
@jan-auer who is best person to do this? should I work on it?